src/Components/Components/src/Dispatcher.cs

```diff
@@ -10,34 +10,46 @@ namespace Microsoft.AspNetCore.Components
     /// <summary>
     /// Dispatches external actions to be executed on the context of a <see cref="Renderer"/>.
     /// </summary>
-    public interface IDispatcher
+    public abstract class Dispatcher
     {
+        /// <summary>
+        /// Provides notifications of unhandled exceptions that occur within the dispatcher.
+        /// </summary>
+        public abstract event UnhandledExceptionEventHandler UnhandledException;
```

Comment:
Do we definitely want this to be public here? Previously I think you could only subscribe to the renderer's `UnhandledException` if you already had a reference to the `Renderer`, which limited it to cases where people were configuring a custom rendering environment. Canonical Blazor apps don't see a reference to a `Renderer` anywhere.

Reply:
We can make this internal.